### PR TITLE
[Overhaul] Implementation of a more consistent & modern UI

### PR DESCRIPTION
**Pull request by KyleKlus**
### Description

This PR is an overhaul of the main ui and it implements the issue [`#872`](https://github.com/st3v3nmw/obsidian-spaced-repetition/issues/872). More detail on what the PR implements and more documentation in general can be found there.

### Changes

-   Improved DeckListView
-   Improved FlashcardReviewView
-   Improved EditModal
-   Added a bunch of classes for styling
-   Renamed gui files such that they follow a uniform naming convention
-   Improved mobile  view
-   Fixes key listener bug from issue #907 
-   Fixes bug from issue #773

### CODE REVIEW

File: manifest.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
     "id": "obsidian-spaced-repetition",
     "name": "Spaced Repetition",
-    "version": "1.11.2",
+    "version": "1.11.1",
```
Comment (st3v3nmw) [inline manifest.json:4]: Looks like there's some conflict here since we shouldn't we're undoing changes that have already gone in

**Review by st3v3nmw — CHANGES_REQUESTED** [review]
Hi, thanks for the contribution! Looks like there are some conflicts since some of the changes are reverting things that have already gone in

File: src/gui/FlashcardModal.tsx [hunk 2]
```diff
@@ -37,55 +40,82 @@ export class FlashcardModal extends Modal {
...
+        // Setup base containers
         this.modalEl.style.height = this.settings.flashcardHeightPercentage + "%";
         this.modalEl.style.width = this.settings.flashcardWidthPercentage + "%";
-        this.modalEl.addClass("sr-modal");
+        this.modalEl.setAttribute("id", "sr-modal");
+
+        this.contentEl.addClass("sr-modal-content");
+        // if (Platform.isMobile) {
+        //     this.contentEl.style.display = "block";
+        // }
```
Comment (st3v3nmw) [inline src/gui/FlashcardModal.tsx:57]: we can delete this if it's no longer required

File: manifest.json (hunk 1 above)
Comment (KyleKlus) [reply manifest.json:4]: I am super sorry, that I caused this (I also read that this cause a mix up with my previous commit). It must have happened, while I was integrating the new commits from your master branch and as I didn't modify these files by hand, I maybe wasn't particularly looking out for mix ups in them 😕.

File: src/gui/FlashcardModal.tsx (hunk 2 above)
Comment (KyleKlus) [reply src/gui/FlashcardModal.tsx]: Yes, I must have not seen it while deleting the commented out code. Its now gone.

File: manifest.json (hunk 1 above)
Comment (KyleKlus) [reply manifest.json]: All is back to what it was before.

**Review by st3v3nmw — APPROVED** [review]
LGTM, thanks.

Please resolve the merge conflicts (caused by merging #902)

**Comment by KyleKlus** [comment]
I think this PR also fixes the bug in issue #768, as I've changed the mobile view such that this shouldn't be a problem anymore, but I don't have a IOS device, so I can't test it.

**Comment by KyleKlus** [comment]
I think this PR also fixes the bug in issue #768, as I've change the mobile view such that this shouldn't be a problem anymore, but I don't have a IOS device, so I can't test it.

**Comment by SimoAbjaou** [comment]
Hello, can you please see this ticket : https://github.com/st3v3nmw/obsidian-spaced-repetition/issues/1101
Can you help me the fix it ?

**Comment by KyleKlus** [comment]
Sry I think I cant help you here, but I haven't changed anything in regards with how the markdown is rendered. I just transformed the containers around the markdown rendered content. Thus I couldn't have changed anything how an Image is rendered.

**Pull request merged**